### DISCUSSION TITLE
I looked at optimization and have a question

### DISCUSSION THREAD

**Discussion by Valinakova** [Hoimar/Planet-Generator#16] [General]
Would rewriting to a lower level language speed up performance substantially?  For example, would the frame rates increase substantially? Would things load much faster?  Or would the benefits be more modest.

 _rewrite core parts in C#? (as parallel branch?)
 rewrite core parts as GDNative module with C or C++? (as parallel branch?)_

**Comment by Hoimar** [comment]
There are two possible performance bottlenecks.
1. **Graphics (rendering):** Would not benefit from a lower level language, this is bottlenecked by your GPU. On my Radeon RX 580 Godot is able to render hundreds of frames per second.
2. **CPU (application logic, terrain generation):** Terrain generation in GDScript is a bottleneck, but does not impact performance of the game as it is multi-threaded. But terrain generation would definitely get a speed boost from a lower level/faster language. I'm not sure which is more suited, C, C++ or C#. The latter one would be slower than the former two though. 

My guess is that it's your GPU.
* can you play around with atmosphere on/off, planet resolution high/low and report back?
* can you check out the `dev` branch of this repository? It has a less expensive atmosphere shader thanks to @creative-brain's help.

**Reply by Valinakova** [reply]
I see... 

On an earlier thread, I mentioned that it sometimes takes sometimes 20 seconds to load a terrain with properly formed detail and this was my experience in game.  For you, I believed you said about 2 seconds.  So judging by what you are saying, it seems a lower language have an impact on this to substantial degree wouldn't it. So maybe instead of 20 seconds and 2 seconds, maybe it would be 4 seconds and 0.5 seconds?
